components/FormularAddRecipe.js

```diff
@@ -0,0 +1,197 @@
+  const [ingredients, setIngredients] = useState([]);
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    onSubmit({ ...data, ingredients });
+  }
+
+  return (
+    <StyledDiv>
+      <StyledForm
+        aria-labelledby={formName}
+        id="recipeForm"
+        onSubmit={handleSubmit}
+      >
+        <StyledInput>
+          <label htmlFor="title">Title</label>
+          <input
+            id="title"
+            name="title"
+            type="text"
+            defaultValue={defaultData?.title}
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="preparation">
+            preparation:
+            <select name="preparation" defaultValue="Oven">
+              <option value="Oven">Oven</option>
+              <option value="Microwave">Microwave</option>
+              <option value="Stove">Stove</option>
+              <option value="Grill">Grill</option>
+            </select>
+          </label>
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="course" defaultValue="Cake">
+            Course:
+            <select name="course" defaultValue="Cake">
+              <option value="Cake">Cake</option>
+              <option value="Dish">Dish</option>
+              <option value="Soup">Soup</option>
+              <option value="Salad">Salad</option>
+              <option value="Cookies">Cookies</option>
+              <option value="Drinks">Drinks</option>
+              <option value="Fingerfood">Fingerfood</option>
+            </select>
+          </label>
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="time">Time:</label>
+          <input
+            id="time"
+            name="time"
+            type="text"
+            defaultValue={defaultData?.time}
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="servings">Servings:</label>
+          <input
+            defaultValue="1"
+            id="servings"
+            name="servings"
+            type="number"
+            min="1"
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="instruction">Instuction</label>
+          <textarea
+            name="instruction"
+            id="instruction"
+            cols="40"
+            rows="10"
+            defaultValue={defaultData?.instruction}
+          ></textarea>
+        </StyledInput>
+      </StyledForm>
+
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          setIngredients([
+            ...ingredients,
+            {
+              ingredientId: nanoid(),
+              name: event.target.elements.ingredient.value,
+              amount: event.target.elements.amount.value,
+              unit: event.target.elements.unit.value,
+            },
+          ]);
+        }}
+      >
+        {" "}
```

Comment:
`Line 148` VSC makes this sometimes automatically, but it does not serve a purpose. (I would remove it)